app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SessionImpl.java

```diff
@@ -472,4 +475,19 @@ public void onLoginUsed(@NonNull PasswordForm form) {
     public WResult<Boolean> checkLoginIfAlreadySaved(PasswordForm form) {
        return mRuntime.getUpLoginPersistence().checkLoginIfAlreadySaved(form);
     }
+
+    @NonNull
+    @Override
+    public UrlUtilsVisitor getUrlUtilsVisitor() {
+        if (mUrlUtilsVisitor == null) {
```

Comment:
Nit: I assume the most likely case is going to be mUrlUtilsVisitor != null, hence perhaps it would be better to early return in that case. What do you think ?